qubesbuilder/plugins/source_rpm/scripts/prepare-chroot-builder

```diff
@@ -0,0 +1,173 @@
+        USER_UID=1000
+    fi
+    if [ -z "$USER_GID" ] && [ -n "$SUDO_GID" ]; then
+        USER_GID="$SUDO_GID"
+    elif [ -z "$USER_GID" ]; then
+        USER_GID=1000
+    fi
+    existing_gid=$(chroot "$INSTALL_DIR" id -g user 2>/dev/null || :)
+    if [ -z "$existing_gid" ]; then
+        chroot "$INSTALL_DIR" groupadd -g "$USER_GID" user
+    elif [ "$existing_gid" != "$USER_GID" ]; then
+        chroot "$INSTALL_DIR" groupmod -g "$USER_GID" user
+    fi
+    existing_uid=$(chroot "$INSTALL_DIR" id -u user 2>/dev/null || :)
+    if [ -z "$existing_uid" ]; then
+        chroot "$INSTALL_DIR" useradd -g user -u "$USER_UID" user
+    elif [ "$existing_uid" != "$USER_UID" ]; then
+        chroot "$INSTALL_DIR" usermod -u "$USER_UID" user
+    fi
+
+    chroot "$INSTALL_DIR" sh -c "chown -R user /home/user;su -c 'mkdir -p qubes-src' - user"
+fi
+
+cp "${PLUGIN_DIR}/repos/builder-local.repo" "$INSTALL_DIR/etc/yum.repos.d/"
+sed -i -e "s#ROOT#$PWD#" "$INSTALL_DIR"/etc/yum.repos.d/*-local.repo
+
+if [ -n "$USE_QUBES_REPO_VERSION" ]; then
+    cp "${PLUGIN_DIR}/repos/qubes-repo-${PACKAGE_SET}-${DIST_NAME}.repo" "$INSTALL_DIR/etc/yum.repos.d/"
+
+    if [ "x$QUBES_MIRROR" != "x" ]; then
+        sed -i "s#baseurl.*yum.qubes-os.org#baseurl = $QUBES_MIRROR#" \
+            "$INSTALL_DIR"/etc/yum.repos.d/qubes-repo-*.repo
+    fi
+
+    sed -i -e "s#%DIST%#$DIST#" "$INSTALL_DIR"/etc/yum.repos.d/qubes-repo-*.repo
+    sed -i -e "s#%QUBESVER%#$USE_QUBES_REPO_VERSION#" "$INSTALL_DIR"/etc/yum.repos.d/qubes-repo-*.repo
+
+    if [ "${DIST_NAME}" = "fedora" ]; then
+        QUBES_SIGNING_KEY="RPM-GPG-KEY-qubes-$USE_QUBES_REPO_VERSION-primary"
+    fi
+    if [ "${DIST_NAME}" = "centos-stream" ]; then
+        QUBES_SIGNING_KEY="RPM-GPG-KEY-qubes-$USE_QUBES_REPO_VERSION-centos"
+    fi
+
+    if [ -r "${PLUGIN_DIR}/keys/$QUBES_SIGNING_KEY" ]; then
+        cp "${PLUGIN_DIR}/keys/$QUBES_SIGNING_KEY" "$INSTALL_DIR/etc/pki/rpm-gpg/"
+        chroot "$INSTALL_DIR" rpm --import "/etc/pki/rpm-gpg/$QUBES_SIGNING_KEY" || true
+    fi
+
+    if [ "0$USE_QUBES_REPO_TESTING" -gt 0 ]; then
+        chroot "$INSTALL_DIR" dnf config-manager --set-enabled 'qubes-builder-*-current-testing'
+    fi
+fi
+
+if ! [ -r "$INSTALL_DIR/proc/cpuinfo" ]; then
+    prepare_chroot_proc
+fi
+
+# remove systemd-resolved symlink
+rm -f "$INSTALL_DIR/etc/resolv.conf"
+cp /etc/resolv.conf "$INSTALL_DIR/etc/"
+# some packages assumes existence of this file
+touch "$INSTALL_DIR"/etc/fstab
+chmod 644 "$INSTALL_DIR"/etc/fstab
+
+echo "INFO: Installing build base packages..."
+
+if [ "$DIST_NAME" = "centos-stream" ]; then
+    if [ "${DIST_VER}" -eq 8 ]; then
+        chroot "$INSTALL_DIR" dnf --disablerepo='qubes*' --releasever="$DIST_VER" copr enable -y fepitre/epel-8-qubes
+        chroot "$INSTALL_DIR" dnf config-manager --disable --releasever="$DIST_VER" '*modular'
+    fi
+fi
+
+# shellcheck disable=SC2086,SC2046
+dnf "${DNF_OPTS[@]}" install -y $(cat ${PKG_LIST_FILE})
```

Comment:
```suggestion
read -r -a PKG_LIST < "${PKG_LIST_FILE}"
dnf "${DNF_OPTS[@]}" install -y -- "${PKG_LIST[@]}"
```

This is more robust and faster, at least if I used `read` correctly.

Reply:
It does not work as is because it contains new lines as separator. Instead of fighting with IFS, `readarray` does the job.